### ISSUE TITLE
Compilation fails if JavaScript files are on a different volume than the Closure Compiler

### ISSUE THREAD

**Issue by Thunderforge** [google/closure-compiler#2589]
If the Closure Compiler is running on one volume (say, the C drive) and your JavaScript files are on a different drive (say, the D drive), compilation will fail with the following exception:

```
java.lang.IllegalArgumentException: 'other' has different root
```

The source of the error is [CompileTask, line 660](https://github.com/google/closure-compiler/blob/master/src/com/google/javascript/jscomp/ant/CompileTask.java#L660), where it tries to call [`Path.relativize(Path other)`](https://docs.oracle.com/javase/8/docs/api/java/nio/file/Path.html#relativize-java.nio.file.Path-) on `Paths.get("")`, which gets the current working directory. Unfortunately, the current working directory is on a different volume, and the above error is thrown. Tested on Windows 7.

Would it be possible to either create a different algorithm for finding the paths, or allow the current working directory to be overridden instead of dependent on `Paths.get("")`?

**Comment by shicks** [comment]
Is changing to the directory of your source files before invoking the compiler a reasonable workaround?  If so, then it seems like the easiest thing to do for starters would be to improve the error message to recommend that.  We don't use the compiler from Windows really ever, so this is unlikely to be fixed without an external PR.

Beyond that, I'm hesitant to add a flag for this.  Do you have a different algorithm in mind?  You're welcome to contribute a fix if you can come up with something that works better across platforms.

**Comment by ChadKillingsworth** [comment]
I use the compiler on windows and usually maintain fixes - but this is such a specific edge case with known workarounds that it won't be a priority for me either.

**Comment by webalchemist** [comment]
Yes, this does appear to be an edge case.  My google search returned this page as the only mention of this error.  

I just installed a new SSD for my operating system and programs and moved my data (including project source code) to a separate drive - then I hit this!  I had closure compiler nicely integrated into my automated build process on my old drive setup too.

I'm guessing there hasn't been a fix proposed for this?

**Comment by lauraharker** [comment]
No, we still don't have a proposed fix.

Created internal issue http://b/122464581
